include/z64environment.h

```diff
@@ -0,0 +1,131 @@
+    /* 1 */ LIGHTNING_STRIKE_START, // fade in the flash. note: bolts are requested in the previous state
+    /* 2 */ LIGHTNING_STRIKE_END // fade out the flash and go back to wait
+} LightningStrikeState;
+
+typedef enum {
+    /*  0 */ SKYBOX_DMA_INACTIVE,
+    /*  1 */ SKYBOX_DMA_FILE1_START,
+    /*  2 */ SKYBOX_DMA_FILE1_DONE,
+    /*  3 */ SKYBOX_DMA_PAL1_START,
+    /* 11 */ SKYBOX_DMA_FILE2_START = 11,
+    /* 12 */ SKYBOX_DMA_FILE2_DONE,
+    /* 13 */ SKYBOX_DMA_PAL2_START
+} SkyboxDmaState;
+
+typedef struct {
+    /* 0x00 */ u8 state;
+    /* 0x01 */ u8 flashRed;
+    /* 0x02 */ u8 flashGreen;
+    /* 0x03 */ u8 flashBlue;
+    /* 0x04 */ u8 flashAlphaTarget;
+    /* 0x08 */ f32 delayTimer;
+} LightningStrike;
+
+// describes what skybox files and blending modes to use depending on time of day
+typedef struct {
+    /* 0x00 */ u16 startTime;
+    /* 0x02 */ u16 endTime;
+    /* 0x04 */ u8 blend; // if true, blend between.. skyboxes? palletes?
+    /* 0x05 */ u8 skybox1Index; // whats the difference between _pal and non _pal files?
+    /* 0x06 */ u8 skybox2Index;
+} struct_8011FC1C; // size = 0x8
+
+typedef struct {
+    /* 0x00 */ u8 ambientColor[3];
+    /* 0x03 */ s8 light1Dir[3];
+    /* 0x06 */ u8 light1Color[3];
+    /* 0x09 */ s8 light2Dir[3];
+    /* 0x0C */ u8 light2Color[3];
+    /* 0x0F */ u8 fogColor[3];
+    /* 0x12 */ s16 fogNear;
+    /* 0x14 */ s16 fogFar;
+} EnvLightSettings;
+
+// 1.0: 801D8EC4
+// dbg: 80222A44
+typedef struct {
+    /* 0x00 */ char unk_00[0x02];
+    /* 0x02 */ u16 timeIncrement; // how many units of time that pass every update
+    /* 0x04 */ Vec3f sunPos; // moon position can be found by negating the sun position
+    /* 0x10 */ u8 skybox1Index;
+    /* 0x11 */ u8 skybox2Index;
+    /* 0x12 */ char unk_12[0x01];
+    /* 0x13 */ u8 skyboxBlend;
+    /* 0x14 */ char unk_14[0x01];
+    /* 0x15 */ u8 skyboxDisabled;
+    /* 0x16 */ u8 sunMoonDisabled;
+    /* 0x17 */ u8 unk_17; // currentWeatherMode for skybox? (prev called gloomySky)
+    /* 0x18 */ u8 unk_18; // nextWeatherMode for skybox?
+    /* 0x19 */ u8 unk_19;
+    /* 0x1A */ u16 unk_1A;
+    /* 0x1C */ char unk_1C[0x02];
+    /* 0x1E */ u8 indoors; // when set, day time has no effect on lighting
+    /* 0x1F */ u8 unk_1F; // outdoor light index
+    /* 0x20 */ u8 unk_20; // prev outdoor light index?
+    /* 0x21 */ u8 unk_21;
+    /* 0x22 */ u16 unk_22;
+    /* 0x24 */ u16 unk_24;
+    /* 0x26 */ char unk_26[0x02];
+    /* 0x28 */ LightInfo dirLight1; // used for sunlight outdoors
+    /* 0x36 */ LightInfo dirLight2; // used for moonlight outdoors
+    /* 0x44 */ s8 skyboxDmaState;
+    /* 0x48 */ DmaRequest dmaRequest;
+    /* 0x68 */ OSMesgQueue loadQueue;
+    /* 0x80 */ OSMesg loadMsg;
+    /* 0x84 */ f32 unk_84;
+    /* 0x88 */ f32 unk_88;
+    /* 0x8C */ s16 adjAmbientColor[3];
+    /* 0x92 */ s16 adjLight1Color[3];
+    /* 0x98 */ s16 adjFogColor[3];
```

Comment:
An enum or macro would be nice for each color in the array.

Reply:
i dont agree, its pretty obvious that rgba is in that order. we have array colors all over the codebase not just here